internal/bzlmod/go_mod.bzl

```diff
@@ -240,16 +326,41 @@ def sums_from_go_mod(module_ctx, go_mod_label):
     """
     _check_go_mod_name(go_mod_label.name)
 
-    # We go through a Label so that the module extension is restarted if go.sum
+    return parse_sumfile(module_ctx, go_mod_label, "go.sum")
+
+def sums_from_go_work(module_ctx, go_work_label):
+    """Loads the entries from a go.work.sum file given a go.work label.
+
+    Args:
+        module_ctx: a https://bazel.build/rules/lib/module_ctx object
+            passed from the MODULE.bazel call.
+        go_work_label: a Label for a `go.work` file. This label is used
+            to find the associated `go.work.sum` file.
+
+    Returns:
+        A Dict[(string, string) -> (string)] is returned where each entry
+        is defined by a Go Module's sum:
+            (path, version) -> (sum)
+    """
+    _check_go_work_name(go_work_label.name)
+
+    return parse_sumfile(module_ctx, go_work_label, "go.work.sum")
+
+def parse_sumfile(module_ctx, label, sumfile):
+    # We go through a Label so that the module extension is restarted if the sumfile
     # changes. We have to use a canonical label as we may not have visibility
-    # into the module that provides the go.sum.
-    go_sum_label = Label("@@{}//{}:{}".format(
-        go_mod_label.workspace_name,
-        go_mod_label.package,
-        "go.sum",
+    # into the module that provides the sumfile
+    sum_label = Label("@@{}//{}:{}".format(
+        label.workspace_name,
+        label.package,
+        sumfile,
     ))
-    go_sum_content = module_ctx.read(go_sum_label)
-    return parse_go_sum(go_sum_content)
+
+    if module_ctx.path(sum_label).exists:
```

Comment:
Does this work if the file doesn't exist? Maybe delete the `go.work.sum` file from the test module as it existing should be the less likely case.

Reply:
@fmeum you are right that doesn't work as I expected. I'm actually not 100% sure how to handle this case now, beyond creating an executable and shelling out to test the files existence which seems funky.

It is really unfortunate to not support both cases \w go.work.sum and without go.work.sum... any guidance?

Reply:
`ctx.path(label)` fails if the file doesn't exist, but `ctx.path(string)` doesn't. So what you can do is:
1. Use `ctx.path()` to resolve the label of go.work to a path. It always exists, so that's fine.
2. Manipulate the path string to be that of `go.work.sum` and turn that into a path, then test for existence.
3. If it exists, also `ctx.path` the label of `go.work.sum` to let Bazel track its contents.
4. If `ctx.watch` is available, also watch the (non-existent) path so that the rule is invalidated if the file shows up. I don't think that's possible without `watch`, but maybe that's fine because the file can't change without any of the `*.mod` files changing?

Reply:
@fmeum oh let me give that a try.

Reply:
the attempt -> https://github.com/bazelbuild/bazel-gazelle/pull/1731